src/pushcollector/_impl/proxy.py

```diff
@@ -50,11 +50,45 @@ class CollectorProxy(object):
     def __init__(self, delegate):
         self._delegate = delegate
 
+    def _translate_pushitem(self, pushitem):
+        if isinstance(pushitem, dict):
+            return [pushitem]
+
+        pushitems = []
+        checksums = {}
+        if pushitem.md5sum:
+            checksums["md5"] = pushitem.md5sum
+        if pushitem.sha256sum:
+            checksums["sha256"] = pushitem.sha256sum
+
+        push_item = {
+            "filename": pushitem.name,
+            "state": pushitem.state,
+            "src": pushitem.src,
+            "dest": None,
+            "checksums": checksums if checksums else None,
+            "origin": pushitem.origin,
+            "build": pushitem.build,
+            "signing_key": pushitem.signing_key,
+        }
+
+        if pushitem.dest:
+            for dest in pushitem.dest:
+                push_item_copy = push_item.copy()
+                push_item_copy["dest"] = dest
+                pushitems.append(push_item_copy)
+
```

Comment:
The logic here all looks right, probably could have benefited from a comment or two though.

We've just had a discussion in code review so it's fresh in our minds, but for someone reading this in a different context it doesn't seem obvious that the intended behavior here is:

- If object has multiple dests, we return one dict per dest.
- If object has no dest at all, we return one dict with an empty dest.

Without a comment mentioning this then one might start to wonder e.g. "is this really intended - why not return an empty list instead for no dest?"